xdsl/dialects/fsm.py

```diff
@@ -45,6 +47,11 @@
 signlessIntegerLike = ContainerOf(AnyOf([IntegerType, IndexType]))
 
 
+@irdl_attr_definition
+class InstanceType(ParametrizedAttribute, TypeAttribute):
+    name = "fsm.instancetype"
```

Comment:
Could you add some doc of this type, please?

Reply:
Unfortunately I was not able to find any. I managed to define it by basically looking at .cpp and .td files, as the documentation refers to it was "FSM instance type" :(